src/firefly/js/metaConvert/AnalysisUtils.js

```diff
@@ -162,24 +167,147 @@ function dispatchResult(dpType, menu,menuKey,dpId, serDef, analysisActivateFunc)
-        return dpdtMessageWithDownload('Cannot display TAR file, you may only download it', 'Download TAR File', url);
+    if (isUsableDownloadType(ext,ct)) {
+        item= makeDownloadType(url,ext,ct,wrapWithMessage, name);
+    }
+    else if (ext.includes('jpeg') || ext.includes('png') || ext.includes('jpg') || ext.includes('gig') ||
+        isSimpleImageType(ct)) {
+        item= makePngEntry(url, name);
     }
-    else if (ext.includes('pdf')) {
-        return dpdtMessageWithDownload('Cannot display PDF file, you may only download it', 'Download PDF File', url);
+    else if (ext.endsWith('txt') || isPlainTextType(ct)) {
+        item= makeTextEntry(url, name);
     }
-    else if (ext.includes('jpeg') || ext.includes('png') || ext.includes('jpg') || ext.includes('gig')) {
-        return dpdtPNG('Show PNG image',url,'dlt-'+idx);
+    else if (ext.endsWith('yaml') || isYamlType(ct)) {
+        item= makeYamlEntry(url, name);
     }
-    return undefined;
+    if (item) item.contentType= ct;
+    return item;
 }
 
-const isNonAnalysisType= (request) => Boolean(fileExtensionSingleProductAnalysis(request));
\ No newline at end of file
+const downloadExts= [ 'tar', 'pdf', 'json'];
+
+export function isUsableDownloadType(ext='', ct) {
+    if (downloadExts.some( (e) => ext.includes(e))) return true;
+    if (!ct) return false;
+    if (isTarType(ct)) return true;
+    if (isGzipType(ct)) return true;
+    if (isJSONType(ct)) return true;
+    if (isPDFType(ct)) return true;
+}
+
+
+export function makeDownloadType(url,ext,ct,wrapWithMessage,name) {
+    const ctL= ct?.toLowerCase();
+    let downloadItem;
+    if (ext.includes('tar') || isTarType(ctL)) {
+        downloadItem= makeTarEntry(url, name);
+    }
+    else if (ext.includes('pdf') || isPDFType(ct)) {
+        downloadItem= makePdfEntry(url, name);
+    }
+    else if (ext.includes('json') || isJSONType(ct)) {
+        downloadItem= makeJsonEntry(url, name);
+    }
+    else if (ext.includes('gzip') || isGzipType(ctL)) {
+        downloadItem= makeGzipEntry(url,name);
+    }
+    else {
+        downloadItem= makeAnyEntry(url,name);
+    }
+    if (downloadItem) {
+        return wrapWithMessage ? dpdtMessage(downloadItem.message, [downloadItem]) : downloadItem;
+    }
+}
+
+
+export function makePdfEntry(url,name) {
+    return dpdtDownload('Download PDF File', url, 'download-0', 'pdf',
+        {
+            message: 'This is a PDF file. It may be downloaded or opened in another tab',
+            loadInBrowserMsg: 'Open PDF File',
+            dropDownText: name ? `${name} (pdf file)` : undefined,
+        }
+    );
+}
+
+export function makeJsonEntry(url,name) {
+    return dpdtDownload('Download JSON File', url, 'download-0', 'json',
+        {
+            message: 'This is a JSON file. It may be downloaded or opened in another tab',
+            loadInBrowserMsg: 'Open JSON File',
+            dropDownText: name ? `${name} (JSON file)` : undefined,
+        }
+    );
+}
```

Comment:
I noticed that the JSON file is currently opened in the browser when users choose to view it. We have a ViewAsText component that can display the content both raw and formatted.  This might be a good option for viewing JSON.